app/models/work.rb

```diff
@@ -0,0 +1,34 @@
+class Work < ApplicationRecord
+  validates :title, presence: true, uniqueness: true
+  validates :creator, presence: true, length: { in: 5..100 }
+  validates_inclusion_of :category, in: %w(book album movie)
+  validates :publication_year, presence: true
+  validates :description, presence: true
+
+  has_many :votes
+  has_many :users, through: :votes
+
+  def self.albums
+    where(category: 'album')
+  end
+
+  def self.books
+    where(category: 'book')
+  end
+
+  def self.movies
+    where(category: 'movie')
+  end
+
+  def self.highest_rated
+    left_outer_joins(:votes)
+      .group("works.id")
+      .order(Arel.sql("count(votes.id) DESC"))
+  end
+
+  def upvote(user)
+    #Vote.create(user: user, work: self)
+    votes.create(user: user)
```

Comment:
Create will return a Vote regardless of whether it succeeds or fails, its up to you to interpret the data! This is why your upvotes don't add more votes to the total, but the error messaging is incorrect.